### DISCUSSION TITLE
Support of TSDZ8

### DISCUSSION THREAD

**Discussion by mstrens** [emmebrusa/TSDZ2-Smart-EBike-1#120] [General]
First many thanks for sharing this excellent firmware.
I have a question:
Is someone interested in adapting this firmware so it can be used with the TSDZ8 motor that seems better than the TSD22(b)?

Best regards.

**Comment by pepsonEL** [comment]
Yes i am interested

**Comment by kiwisibk** [comment]
Absolutely. That would be fantastic! I can hardly wait...

**Comment by ShalokShalom** [comment]
Do we already know, if it has the same controller?

**Reply by jussisaily** [reply]
Unfortunately it's a different controller, and you need the J-link programmer instead of ST-link used with the TSDZ2. I could not easily find the exact controller type they're using. Open source firmware does not exist (yet) for the TSDZ8.

**Reply by ShalokShalom** [reply]
Yeah, I just found the programmer:

https://ebikestuff.eu/en/content/18-tsdz8-firmware-update-using-j-link-jlink-v9-programming-kit

**Reply by ShalokShalom** [reply]
Someone said its the same controller used on the tsdz2 v2

**Reply by jussisaily** [reply]
It's supposedly the Infineon XMC1302 controller.

**Reply by mstrens** [reply]
It is for sure not the same controller. As said above, it uses a different microprocessor (much more powerful). Still it uses the same other external components ( torque sensor, hall sensors, ...). So someone tried to replace the tsdz8 controller by a tsdz2 controller and it first seems to be ok (to be confirmed by some more tests). Note: the tsdz2 controller did not fit inside the motor of the tsdz8 and so had to be fixed externally.

**Comment by MarcoLauria27** [comment]
It is not the same controller. As said before, it uses a more powerful Infineon microcontroller instead of the STM8S105. It also has better MOSFETs. I've adapted casainho's code from https://github.com/OpenSourceEBike/EBike_EScooter_modular_DIY, running circuitpython on an ESP32 and using a FLIPSKY microntroller to run on the TSDZ8 with the color TSDZ2 display. It's on the early stages because I had not much time to work on it lately. The display interface is working. Haven't fully modified the torque sensor code (the one from casainho uses CANBUS) but that's something that can be easily done. Haven't tested on an actual bike but it's running on the bench with PAS only.

**Reply by ShalokShalom** [reply]
Why dont we program the new controller?

**Reply by MarcoLauria27** [reply]
It can certainly be done but someone would need to map the logic and the ports the sensors are connected. It requires considerable effort. One advantage of using the VESC controller (like the Flipsky mini) is to use the VESC software to calibrate the motor. The ESP32 is very small and fit the controller case but the Flipsky mini doesn't. I was not able to find a smaller VESC compatible board that would fit with the ESP32. Using the ESP32 also allows the use of python and that can be easily modified via BT later if needed.
